src/main/java/com/example/gradecalculator/controller/UserController.java

```diff
@@ -127,5 +125,20 @@ public String performLogout(Authentication authentication, HttpServletRequest re
         this.logoutHandler.logout(request, response, authentication);
         return "index";
     }
+
+    @DeleteMapping("/myProfile")
+    public ResponseEntity<Void> deleteUser(Authentication authentication,HttpServletRequest request, HttpServletResponse response) {
+        var userId = userService.getAuthenticatedUserId(authentication);
+
+        boolean profileDeleted = userService.deleteUserAndSubjects(userId);
+        if (profileDeleted) {
+            performLogout(authentication, request, response);
+            return ResponseEntity.ok().build();
+
+        } else {
+            return ResponseEntity.badRequest().build();
+        }
```

Comment:
Because there is a return, you could also write it like this:
```suggestion
        if (profileDeleted) {
            performLogout(authentication, request, response);
            return ResponseEntity.ok().build();
        }
        
        return ResponseEntity.badRequest().build();

```
Not a must, just an information :)